### PR TITLE
fix lassen debug build failure

### PR DESCRIPTION
**Pull request by rrsettgast**
We were getting errors building debug on Lassen.

```
/usr/WS2/settgast/Codes/geosx/GEOSX_develop/src/coreComponents/LvArray/src/indexing.hpp(194): error: calling a __host__ function("LvArray::ArrayOfSets<int, int,  ::LvArray::ChaiBuffer> ::ArrayOfSets") from a __host__ __device__ function("") is not allowed

/usr/WS2/settgast/Codes/geosx/GEOSX_develop/src/coreComponents/LvArray/src/indexing.hpp(194): error: calling a __host__ function("LvArray::ArrayOfSets<int, int,  ::LvArray::ChaiBuffer> ::ArrayOfSets") from a __host__ __device__ function("") is not allowed

2 errors detected in the compilation of "/var/tmp/settgast/tmpxft_0000884f_00000000-6_EdgeManager.cpp1.ii".
```
changed the lambda to `LVARRAY_HOST_DEVICE`, then this:
```
/usr/WS2/settgast/Codes/geosx/GEOSX_develop/src/coreComponents/LvArray/src/indexing.hpp(190): error: An extended __host__ __device__ lambda cannot capture variables by reference
          detected during:
            instantiation of "void LvArray::indexing::checkIndices(const INDEX_TYPE *, INDICES...) [with INDEX_TYPE=geosx::localIndex, INDICES=<geosx::localIndex>]" 
/usr/WS2/settgast/Codes/geosx/GEOSX_develop/src/coreComponents/LvArray/src/ArrayView.hpp(461): here
            instantiation of "INDEX_TYPE LvArray::ArrayView<T, NDIM_TPARAM, USD_TPARAM, INDEX_TYPE, BUFFER_TYPE>::linearIndex(INDICES...) const [with T=geosx::integer, NDIM_TPARAM=1, USD_TPARAM=0, INDEX_TYPE=geosx::localIndex, BUFFER_TYPE=LvArray::ChaiBuffer, INDICES=<geosx::localIndex>]" 
/usr/WS2/settgast/Codes/geosx/GEOSX_develop/src/coreComponents/LvArray/src/sliceHelpers.hpp(69): here
            instantiation of "void LvArray::forValuesInSliceWithIndices(T &, LAMBDA &&, INDICES...) [with T=int, LAMBDA=lambda [](const geosx::integer &, auto...)->auto, INDICES=<>]" 
/usr/WS2/settgast/Codes/geosx/GEOSX_develop/src/coreComponents/dataRepository/Wrapper.hpp(411): here
            instantiation of "void geosx::dataRepository::Wrapper<T>::copy_wrapper::copy(const geosx::Array<U, NDIM, PERMUTATION> &, geosx::localIndex, geosx::localIndex) [with T=geosx::array1d<geosx::integer>, U=geosx::integer, NDIM=1, PERMUTATION=camp::make_idx_seq_t<1L>]" 
/usr/WS2/settgast/Codes/geosx/GEOSX_develop/src/coreComponents/dataRepository/Wrapper.hpp(435): here
            instantiation of "void geosx::dataRepository::Wrapper<T>::copy(geosx::localIndex, geosx::localIndex) [with T=geosx::array1d<geosx::integer>]" 
/usr/WS2/settgast/Codes/geosx/GEOSX_develop/src/coreComponents/dataRepository/Wrapper.hpp(132): here
            instantiation of "geosx::dataRepository::Wrapper<T>::Wrapper(const geosx::string &, geosx::dataRepository::Group &, T *) [with T=geosx::array1d<geosx::integer>]" 
/usr/WS2/settgast/Codes/geosx/GEOSX_develop/src/coreComponents/dataRepository/Group.hpp(1582): here
            instantiation of "geosx::dataRepository::Wrapper<T> &geosx::dataRepository::Group::registerWrapper(const geosx::string &, T *) [with T=geosx::array1d<geosx::integer>]" 
/usr/WS2/settgast/Codes/geosx/GEOSX_develop/src/coreComponents/mesh/mpiCommunications/NeighborData.hpp(42): here

/usr/WS2/settgast/Codes/geosx/GEOSX_develop/src/coreComponents/LvArray/src/indexing.hpp(190): error: An extended __host__ __device__ lambda cannot capture variables by reference
          detected during:
            instantiation of "void LvArray::indexing::checkIndices(const INDEX_TYPE *, INDICES...) [with INDEX_TYPE=geosx::localIndex, INDICES=<geosx::localIndex>]" 
/usr/WS2/settgast/Codes/geosx/GEOSX_develop/src/coreComponents/LvArray/src/ArrayView.hpp(461): here
            instantiation of "INDEX_TYPE LvArray::ArrayView<T, NDIM_TPARAM, USD_TPARAM, INDEX_TYPE, BUFFER_TYPE>::linearIndex(INDICES...) const [with T=geosx::integer, NDIM_TPARAM=1, USD_TPARAM=0, INDEX_TYPE=geosx::localIndex, BUFFER_TYPE=LvArray::ChaiBuffer, INDICES=<geosx::localIndex>]" 
/usr/WS2/settgast/Codes/geosx/GEOSX_develop/src/coreComponents/LvArray/src/sliceHelpers.hpp(69): here
            instantiation of "void LvArray::forValuesInSliceWithIndices(T &, LAMBDA &&, INDICES...) [with T=int, LAMBDA=lambda [](const geosx::integer &, auto...)->auto, INDICES=<>]" 
/usr/WS2/settgast/Codes/geosx/GEOSX_develop/src/coreComponents/dataRepository/Wrapper.hpp(411): here
            instantiation of "void geosx::dataRepository::Wrapper<T>::copy_wrapper::copy(const geosx::Array<U, NDIM, PERMUTATION> &, geosx::localIndex, geosx::localIndex) [with T=geosx::array1d<geosx::integer>, U=geosx::integer, NDIM=1, PERMUTATION=camp::make_idx_seq_t<1L>]" 
/usr/WS2/settgast/Codes/geosx/GEOSX_develop/src/coreComponents/dataRepository/Wrapper.hpp(435): here
            instantiation of "void geosx::dataRepository::Wrapper<T>::copy(geosx::localIndex, geosx::localIndex) [with T=geosx::array1d<geosx::integer>]" 
/usr/WS2/settgast/Codes/geosx/GEOSX_develop/src/coreComponents/dataRepository/Wrapper.hpp(132): here
            instantiation of "geosx::dataRepository::Wrapper<T>::Wrapper(const geosx::string &, geosx::dataRepository::Group &, T *) [with T=geosx::array1d<geosx::integer>]" 
/usr/WS2/settgast/Codes/geosx/GEOSX_develop/src/coreComponents/dataRepository/Group.hpp(1582): here
            instantiation of "geosx::dataRepository::Wrapper<T> &geosx::dataRepository::Group::registerWrapper(const geosx::string &, T *) [with T=geosx::array1d<geosx::integer>]" 
/usr/WS2/settgast/Codes/geosx/GEOSX_develop/src/coreComponents/mesh/mpiCommunications/NeighborData.hpp(42): here

/usr/WS2/settgast/Codes/geosx/GEOSX_develop/src/coreComponents/LvArray/src/indexing.hpp(190): error: __host__ __device__ extended lambdas cannot be generic lambdas
          detected during:
            instantiation of "void LvArray::indexing::checkIndices(const INDEX_TYPE *, INDICES...) [with INDEX_TYPE=geosx::localIndex, INDICES=<geosx::localIndex>]" 
/usr/WS2/settgast/Codes/geosx/GEOSX_develop/src/coreComponents/LvArray/src/ArrayView.hpp(461): here
            instantiation of "INDEX_TYPE LvArray::ArrayView<T, NDIM_TPARAM, USD_TPARAM, INDEX_TYPE, BUFFER_TYPE>::linearIndex(INDICES...) const [with T=geosx::integer, NDIM_TPARAM=1, USD_TPARAM=0, INDEX_TYPE=geosx::localIndex, BUFFER_TYPE=LvArray::ChaiBuffer, INDICES=<geosx::localIndex>]" 
/usr/WS2/settgast/Codes/geosx/GEOSX_develop/src/coreComponents/LvArray/src/sliceHelpers.hpp(69): here
            instantiation of "void LvArray::forValuesInSliceWithIndices(T &, LAMBDA &&, INDICES...) [with T=int, LAMBDA=lambda [](const geosx::integer &, auto...)->auto, INDICES=<>]" 
/usr/WS2/settgast/Codes/geosx/GEOSX_develop/src/coreComponents/dataRepository/Wrapper.hpp(411): here
            instantiation of "void geosx::dataRepository::Wrapper<T>::copy_wrapper::copy(const geosx::Array<U, NDIM, PERMUTATION> &, geosx::localIndex, geosx::localIndex) [with T=geosx::array1d<geosx::integer>, U=geosx::integer, NDIM=1, PERMUTATION=camp::make_idx_seq_t<1L>]" 
/usr/WS2/settgast/Codes/geosx/GEOSX_develop/src/coreComponents/dataRepository/Wrapper.hpp(435): here
            instantiation of "void geosx::dataRepository::Wrapper<T>::copy(geosx::localIndex, geosx::localIndex) [with T=geosx::array1d<geosx::integer>]" 
/usr/WS2/settgast/Codes/geosx/GEOSX_develop/src/coreComponents/dataRepository/Wrapper.hpp(132): here
            instantiation of "geosx::dataRepository::Wrapper<T>::Wrapper(const geosx::string &, geosx::dataRepository::Group &, T *) [with T=geosx::array1d<geosx::integer>]" 
/usr/WS2/settgast/Codes/geosx/GEOSX_develop/src/coreComponents/dataRepository/Group.hpp(1582): here
            instantiation of "geosx::dataRepository::Wrapper<T> &geosx::dataRepository::Group::registerWrapper(const geosx::string &, T *) [with T=geosx::array1d<geosx::integer>]" 
/usr/WS2/settgast/Codes/geosx/GEOSX_develop/src/coreComponents/mesh/mpiCommunications/NeighborData.hpp(42): here
```

@victorapm This will fix the problem you are having.

https://github.com/GEOSX/LvArray/pull/270

### CODE REVIEW

**Comment by corbett5** [comment]
Fixed it. Trust the error message, not the line numbers I guess haha. I undid the LvArray update too.

File: src/coreComponents/dataRepository/BufferOps_inline.hpp [hunk 1]
```diff
@@ -1560,7 +1560,7 @@ Unpack( buffer_unit_type const * & buffer,
   {
     // Copy old capacities (no way to direct copy, must kernel launch)
     array1d< localIndex > newCapacities( var.size() );
-    forAll< parallelHostPolicy >( var.size(), [var, newCapacities = newCapacities.toView()]( localIndex const k )
```
Comment (klevzoff) [inline src/coreComponents/dataRepository/BufferOps_inline.hpp:1563]: :facepalm: 
This was intended to be `var = var.toViewConst()`

Comment (corbett5) [reply src/coreComponents/dataRepository/BufferOps_inline.hpp:1563]: I like that better than the capture by reference I did.

File: src/coreComponents/constitutive/fluid/PVTDriver.cpp [hunk 2]
```diff
@@ -108,9 +108,9 @@ void PVTDriver::postProcessInput()
 
   for( integer n=0; n<m_numSteps+1; ++n )
   {
-    m_table( n, TIME ) = minTime + n*dt;
-    m_table( n, PRES ) = pressureFunction.evaluate( &m_table( n, TIME ) );
-    m_table( n, TEMP ) = temperatureFunction.evaluate( &m_table( n, TIME ) );
+    m_table( n, static_cast< localIndex >(TIME) ) = minTime + n*dt;
```
Comment (klevzoff) [inline src/coreComponents/constitutive/fluid/PVTDriver.cpp:111]: What was the issue here? Would it be solved if we instead specified `localIndex` as the base type of `columnKeys` enum in `PVTDriver.hpp` and `TriaxialDriver.hpp`?

**Review by klevzoff — APPROVED** [review]

File: src/coreComponents/constitutive/fluid/PVTDriver.cpp (hunk 2 above)
Comment (rrsettgast) [reply src/coreComponents/constitutive/fluid/PVTDriver.cpp:111]: The compiler wasn't recognizing the implicit conversation of the enum to an int. The static_cast made the errors go away, I will test this branch with Ben's fix, without these "workarounds" as I think they shouldn't be necessary.

**Pull request closed without merging**